src/components/molecules/RecordSearchBar/index.tsx

```diff
@@ -0,0 +1,68 @@
+"use client";
+
+import { getCities } from "@/apis/district";
+import Button from "@/components/atoms/Button";
+import Dropdown from "@/components/atoms/Dropdown";
+import { useQuery } from "@tanstack/react-query";
+import { useCallback, useState } from "react";
+import { useRouter, useSearchParams } from "next/navigation";
+import { formatDateToStringByDash } from "@/utils/formatDateToString";
+import formatStringToDateByDash from "@/utils/formatStringToDate";
+import SimpleDatePicker from "../SimpleDatePicker";
+
+function RecordSearchBar(): JSX.Element {
+  const router = useRouter();
+  const searchParams = useSearchParams();
+
+  const currentDate = new Date();
+  const threeMonthsAgoDate = new Date();
+  threeMonthsAgoDate.setMonth(currentDate.getMonth() - 3);
+  const searchParamStartDate = formatStringToDateByDash(searchParams.get("start") as string);
+  const searchParamEndDate = formatStringToDateByDash(searchParams.get("end") as string);
+  const [cityId, setCityId] = useState(0);
+  const [startDate, setStartDate] = useState(searchParamStartDate || threeMonthsAgoDate);
+  const [endDate, setEndDate] = useState(searchParamEndDate || currentDate);
+
+  const { data } = useQuery({
+    queryKey: ["cityId"],
+    queryFn: getCities,
+  });
+  const cities = data?.data?.response?.cities ?? [];
+
+  const handleOnClick = useCallback(() => {
+    const searchParamObj = new URLSearchParams(searchParams);
+    if (cityId) searchParamObj.set("cityId", cityId.toString());
+    else searchParamObj.delete("cityId");
+    searchParamObj.set("start", formatDateToStringByDash(startDate));
+    searchParamObj.set("end", formatDateToStringByDash(endDate));
+    const queryString = searchParamObj.toString();
+    router.push(`?${queryString}`);
+  }, [cityId, endDate, router, searchParams, startDate]);
```

Comment:
코드가 너무 압축되어 있어서 가독성이 떨어지는 것 같아요! 저라면 중간 중간 공백과 생략된 괄호도 넣어줄 것 같고 참고차 보시면 될 것 같습니다!
```suggestion
  const handleOnClick = useCallback(() => {
    const searchParamObj = new URLSearchParams(searchParams);
    
    if (cityId) {
      searchParamObj.set("cityId", cityId.toString());
    } else {
      searchParamObj.delete("cityId");
    }
    
    searchParamObj.set("start", formatDateToStringByDash(startDate));
    searchParamObj.set("end", formatDateToStringByDash(endDate));
    const queryString = searchParamObj.toString();
    
    router.push(`?${queryString}`);
  }, [cityId, endDate, router, searchParams, startDate]);
```

Reply:
수정하여 반영했습니다